apis/python/test/test_distance_metrics.py

```diff
@@ -15,6 +21,193 @@ def normalize_vectors(vectors):
+    np.array([[2, 2, 2, 2]], dtype=np.float32)
+
+    expected_distances = np.array([0.500000, 0.292893, 0.142262, 0.239361, 0.133975])
+
+    query_and_check(
+        index,
+        np.array([[2, 2, 2, 2]], dtype=np.float32),
+        3,
+        {2, 3, 4},
+        expected_distances=expected_distances,
+        nprobe=2,
+    )
+    distances, ids = index.query(
+        queries=np.array([[2, 2, 2, 2]], dtype=np.float32), k=5
+    )
+    assert np.array_equal(ids, np.array([[4, 2, 3, 1, 0]], dtype=np.uint64))
+    sorted_distances = np.sort(distances)
+    assert np.allclose(distances, sorted_distances, 1e-4)
+
+
+def test_ivf_flat_index(capfd, tmp_path):
+    partitions = 10
+    uri = os.path.join(tmp_path, "array")
+    vector_type = np.dtype(np.float32)
+    index = ivf_flat_index.create(
+        uri=uri,
+        dimensions=4,
+        vector_type=vector_type,
+        partitions=partitions,
+        distance_metric=vspy.DistanceMetric.COSINE,
+    )
+
+    ingestion_timestamps, base_sizes = load_metadata(uri)
+    assert base_sizes == [0]
+    assert ingestion_timestamps == [0]
+
+    query_and_check(
+        index,
+        np.array([[2, 2, 2, 2]], dtype=np.float32),
+        3,
+        {MAX_UINT64},
+        nprobe=partitions,
+    )
+
+    update_vectors = np.empty([5], dtype=object)
+    update_vectors[0] = np.array([1, 0, 0, 0], dtype=vector_type)
+    update_vectors[1] = np.array([1, 1, 0, 0], dtype=vector_type)
+    update_vectors[2] = np.array([32, 41, 30, 0], dtype=vector_type)
+    update_vectors[3] = np.array([1, 5, 3, 0], dtype=vector_type)
+    update_vectors[4] = np.array([4, 4, 4, 0], dtype=vector_type)
+    index.update_batch(vectors=update_vectors, external_ids=np.array([0, 1, 2, 3, 4]))
+
+    expected_distances = np.array([0.500000, 0.292893, 0.142262, 0.239361, 0.133975])
+
+    query_and_check(
+        index,
+        np.array([[2, 2, 2, 2]], dtype=np.float32),
+        3,
+        {2, 3, 4},
+        expected_distances=expected_distances,
+        nprobe=partitions,
+    )
+
+    index = index.consolidate_updates()
+
+    query_and_check(
+        index,
+        np.array([[2, 2, 2, 2]], dtype=np.float32),
+        3,
+        {2, 3, 4},
+        expected_distances=expected_distances,
+        nprobe=partitions,
+    )
+
+    distances, ids = index.query(
+        queries=np.array([[2, 2, 2, 2]], dtype=np.float32), k=5
+    )
+    assert np.array_equal(ids, np.array([[4, 2, 3, 1, 0]], dtype=np.uint64))
+    sorted_distances = np.sort(distances)
+    assert np.allclose(distances, sorted_distances, 1e-4)
+
+    # Update the index with a new vector
+    index.update(
+        vector=np.array([2, 2, 2, 2], dtype=vector_type),
+        external_id=5,
+    )
+    expected_distances = np.append(expected_distances, 0.0)
+
+    # delete the vector with external id 3
+    index.delete(external_id=3)
+
+    # consolidate the updates
+    index = index.consolidate_updates()
+
+    # Query the index
```

Comment:
(Minor) Could we also add this to make sure the order is correct even after an `update()` + `delete()`?
```
    distances, ids = index.query(
        queries=np.array([[2, 2, 2, 2]], dtype=np.float32), k=5
    )
    assert np.array_equal(ids, np.array([[5, 4, 2, 1, 0]], dtype=np.uint64))
    sorted_distances = np.sort(distances)
    assert np.allclose(distances, sorted_distances, 1e-4)
```